### PR TITLE
Task/171 Total balance widget in inventory screen

### PR DESCRIPTION
**Pull request by TomasPetrik**
Incremented targetSdkVersion to 32,
Squashed commits.

### CODE REVIEW

File: lib/extensions/number_extensions.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'package:intl/intl.dart';
+
+extension DoubleParseExtension on double {
+  String getThousandSeparatedValue() {
+    NumberFormat numberFormat = NumberFormat("#,##0.00");
+    return numberFormat.format(this);
+  }
+
+  String getThousandValueWithCurrency({
+    String currencySymbol = '\$',
```
Comment (jvrastil) [inline lib/extensions/number_extensions.dart:10]: do we plan to support only USD? @daxicek

File: android/app/build.gradle [hunk 2]
```diff
@@ -50,7 +50,7 @@ android {
```
Comment (jvrastil) [inline android/app/build.gradle:50]: there is an issue with missing bottom bar when dropdown is opened

image
image

**Review by jvrastil — CHANGES_REQUESTED** [review]

File: android/app/build.gradle (hunk 2 above)
Comment (TomasPetrik) [reply android/app/build.gradle:50]: Fixed.

File: lib/extensions/number_extensions.dart (hunk 1 above)
Comment (daxicek) [reply lib/extensions/number_extensions.dart:10]: No, list of currencies is in system-assets endpoint.

Comment (TomasPetrik) [reply lib/extensions/number_extensions.dart:10]: Ok, so this specific currency may be passed to currencySymbol as soon as we will have it defined in our data objects.

File: lib/view/screens/discovery_screen.dart [hunk 3]
```diff
@@ -10,13 +10,15 @@ class DiscoveryScreen extends StatelessWidget {
...
-        subtitle: 'User Level 1',
-      ),
-      SliverToBoxAdapter(
-        child: Container(
+    return CustomScrollView(
+      physics: const BouncingScrollPhysics(),
+      slivers: [
+        const SioAppBar(
+          title: 'Nick name',
+          subtitle: 'User Level 1',
```
Comment (ciripel) [inline lib/view/screens/discovery_screen.dart:18]: shouldn't the level be dynamic?

**Review by ciripel — APPROVED** [review]

File: lib/view/screens/discovery_screen.dart (hunk 3 above)
Comment (TomasPetrik) [reply lib/view/screens/discovery_screen.dart:18]: @ciripel 
Yes, you are right but this SioAppBar with leveling will be done in separated task.

File: lib/extensions/number_extensions.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import 'package:intl/intl.dart';
```
Comment (storkandstars) [inline lib/extensions/number_extensions.dart:1]: Move this extension folder to a relative domain. This specific extension will be used only inside the view layer. 
A. If it is specific to widgets. Move it to `widgets/extensions`
B. If it is specific to screens. Move it to `screens/extensions`
C. If it is specific to view in general. Move it to `view/extensions`. In that case consider if it is related to routes, themes, etc.

Each layer might have their mixins, extensions and other reusable components.

File: lib/extensions/number_extensions.dart [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  String getThousandValueWithCurrency({
+    String currencySymbol = '\$',
+    CurrencySide currencySide = CurrencySide.left,
+  }) {
+    String formattedValue = getThousandSeparatedValue();
+
+    if (currencySide == CurrencySide.left) {
+      return '$currencySymbol$formattedValue';
+    } else {
```
Comment (storkandstars) [inline lib/extensions/number_extensions.dart:17]: Tip: Omit unnecessary `else` when using `return` in previous condition block. It reduces nested code.

File: lib/extensions/number_extensions.dart [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  String getThousandSeparatedValue() {
+    NumberFormat numberFormat = NumberFormat("#,##0.00");
+    return numberFormat.format(this);
+  }
+
+  String getThousandValueWithCurrency({
+    String currencySymbol = '\$',
+    CurrencySide currencySide = CurrencySide.left,
+  }) {
+    String formattedValue = getThousandSeparatedValue();
```
Comment (storkandstars) [inline lib/extensions/number_extensions.dart:13]: Allow ability to re-assignment value only when it is desired otherwise prefer making values `final`.

**Review by storkandstars — CHANGES_REQUESTED** [review]

File: lib/extensions/number_extensions.dart (hunk 4 above)
Comment (TomasPetrik) [reply lib/extensions/number_extensions.dart:1]: Okay. Moved it to view/extensions.

File: lib/extensions/number_extensions.dart (hunk 6 above)
Comment (TomasPetrik) [reply lib/extensions/number_extensions.dart:13]: Fixed by using final.

**Review by storkandstars — APPROVED** [review]

**Review by storkandstars — APPROVED** [review]

**Pull request merged**